We can now build from anywhere

maurolepore opened the pull request:
This commit makes the command portable: we can call it from anywhere.
The location of the command is used to find the Dockerfile and to
output the image's .tar.gz file.

I also updated README to reflect the changes (plus a lil style -- happy to submit separately if neede).

maurolepore committed:
We can now build from anywhere

maurolepore committed:
We can now build from anywhere

maurolepore committed:
Update README to reflect latest changes

maurolepore committed:
We can now build from anywhere

This commit makes the command portable: we can call it from anywhere.
The location of the command is used to find the Dockerfile and to
output the image's .tar.gz file.

maurolepore committed:
We can now build from anywhere

This commit makes the command portable: we can call it from anywhere.
The location of the command is used to find the Dockerfile and to
output the image's .tar.gz file.

maurolepore committed:
merge origin/no-longer-need-to-run-from-pacta

maurolepore commented:
I built this with internet form my phone so I didn't fully test it. No rush -- I can test tomorrow.

maurolepore committed:
merge origin/no-longer-need-to-run-from-pacta

cjyetman commented on pacta/README.md:
```diff
@@ -29,24 +31,28 @@ Run the build_with_tag.sh script from this directory, specifying a tag to assign
+- build a "2dii_pacta" docker image using the Dockerfile in this directory,
+which will
   - use 2dii/r-packages as a base
   - copy in the frshly cloned repos
```
freshly

cjyetman commented on pacta/README.md:
```diff
@@ -29,24 +31,28 @@ Run the build_with_tag.sh script from this directory, specifying a tag to assign
   - copy in the frshly cloned repos
   - make some necessary permissions changes
-- use pacta_web_template.zip as a template to create new zip file named pacta_web.zip in this directory which will contain
   - an export of the freshly made docker image gzipped (2dii_pacta.tar.gz)
```
start this point with a verb, e.g. "export the..." or "make an export of the..."

cjyetman commented on pacta/README.md:
```diff
@@ -29,24 +31,28 @@ Run the build_with_tag.sh script from this directory, specifying a tag to assign
-  - a few scripts and directories that contain sample data to facilitate testing the new image
 
-If the build is succesful, one should test it with the test scripts, and if the tests are successful, load the image interactively and push the tags created for each of the PACTA_analysis and friends repos inside of the docker container.
+If the build is succesful, one should test it with the test scripts, and if
```
do we still have test scripts here?

cjyetman reviewed (commented):
how does it find the `Dockerfile` file?

cjyetman commented:
we should also mention in the Readme that this adds both the tag that you assign, plus "latest"... and *maybe* something about how that enables you to call the docker image without needing to specify the precise tag

maurolepore commented on pacta/build_with_tag.sh:
```diff
@@ -76,7 +69,8 @@ do
 done
 
-image_tar_gz="2dii_pacta.tar.gz"
+parent="$(dirname $(which $0))"
```
This is how we find the Dockerfile:

The command and Dockerfile need to live in the same directory (we might relax this eventually, by adding an argument `--path`). The special argument `$0` contains the command the user typed -- i.e. `*build_with_tag.sh`. Then we use `which` to get the full path. Finally we extract the dirname (e.g. `~/git/2diidockerrunner/pacta`), and there we now we have the Dockerfile.

cjyetman replied:
fancy! 👍

maurolepore committed:
Fix typo

maurolepore committed:
Remove reference to test scripts

maurolepore committed:
Start a list item with a verb

maurolepore committed:
Remove reference to 'run from anywhere'

Users can lookup usage in the documentation of the command itself.
Explaining that here is duplicating information that make it harder
to maintain -- i.e. keep in sync with what the program actually
does.

maurolepore committed:
Explain double-tagging as 'latest'

maurolepore commented:
I'm made the changes you proposed.

To think and maybe follow up in another PR:

I'm a bit uneasy about  README because I think it will be hard to maintain. It exposes implementation details that are likely to change. I would prefer to keep README higher level, and document the implementation details with more evocative code, and/or, with comments next to the code, and/or with a help file at the top of the script itself.

cjyetman commented on pacta/README.md:
```diff
@@ -29,17 +29,18 @@ previously existing tags in the PACTA_analysis and friends repos.
 
-If the build is succesful, one should test it with the test scripts, and if
-the tests are successful, load the image interactively and push the tags
+If the build is succesful, load the image interactively and push the tags
```
succesful -> successful

maurolepore committed:
Finally use a spellcheck :)

cjyetman reviewed (approved):
we need an editor! hahaha

lgtm
